compiler/rustc_passes/src/stability.rs

```diff
@@ -90,6 +90,26 @@ fn inherit_deprecation(def_kind: DefKind) -> InheritDeprecation {
     }
 }
 
+fn inherit_const_stability(tcx: TyCtxt<'_>, def_id: LocalDefId) -> InheritConstStability {
+    let def_kind = tcx.def_kind(def_id);
+    match def_kind {
+        DefKind::AssocFn | DefKind::AssocTy | DefKind::AssocConst => {
+            let hir_id = tcx.hir().local_def_id_to_hir_id(def_id);
+            let item = tcx.hir().expect_item(tcx.hir().get_parent_item(hir_id));
+            match item.kind {
+                hir::ItemKind::Impl(hir::Impl { of_trait: Some(_), .. }) => {
+                    InheritConstStability::Yes
+                }
+                _ => InheritConstStability::No,
+            }
+        }
+        DefKind::Field | DefKind::Variant | DefKind::Ctor(CtorOf::Struct, _) => {
+            InheritConstStability::Yes
+        }
```

Comment:
why are these three `Yes`? It seems to me like these shouldn't have const stability attributes

Reply:
I don't know enough of const stability semantics. I tried to reproduce the existing. I probably have missed some things because the logic for const stability was inverted.